clusterloader2/pkg/measurement/common/api_availability_measurement.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"k8s.io/klog"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+	"time"
+)
+
+const (
+	apiAvailabilityName = "ApiAvailability"
+)
+
+func init() {
+	if err := measurement.Register(apiAvailabilityName, createApiAvailabilityMeasurement); err != nil {
+		klog.Fatalf("Cannot register %s: %v", apiAvailabilityName, err)
+	}
+}
+
+func createApiAvailabilityMeasurement() measurement.Measurement {
+	return &apiAvailabilityMeasurement{}
+}
+
+type apiAvailabilityMeasurement struct{}
+
+// Execute starts the api-server healthz probe end point from start action and
+// collect metrics in gather.
+func (a apiAvailabilityMeasurement) Execute(config *measurement.MeasurementConfig) ([]measurement.Summary, error) {
+	action, err := util.GetString(config.Params, "action")
+	if err != nil {
+		return nil, err
+	}
+	probeDuration, err := util.GetInt(config.Params, "duration")
+	if err != nil {
+		return nil, err
+	}
+	ticker := time.NewTicker(500 * time.Millisecond)
+	stopAvailabilityCheck := make(chan bool)
+	apiServerClient := config.ClusterFramework.GetClientSets().GetClient().CoreV1().RESTClient()
+	numProbes := 0
```

Comment:
Instead of having one huge Execute method, I would much rather prefer having a more organized approach with dedicate start() and gather() methods, those variables being measurement fields, etc.

As an example, PTAL at profile measurement, which is pretty similar conceptually to what you want to achieve:
https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/pkg/measurement/common/profile.go

Reply:
thanks @wojtek-t this was just an initial stab. That's why I created a draft PR to get comments and the direction i should be moving in.

Reply:
@wojtek-t thanks for pointing me to this example. As you said, it's pretty similar to what I want to achieve.
Couple of questions about interacting with api server. I see that there's explicit configs to interact with https and also add rbac roles to expose debug end point. Do you think we need all those features in this measurement?

Reply:
> I  see that there's explicit configs to interact with https and also add rbac roles to expose debug end point. Do you think we need all those features in this measurement?

https is definitely needed. Re rbac - I can't remember from the top of my head how /health is handled - I think it's not needed, but it requires checking once you implement the measurement itself.

Reply:
ok. Another question that I have is, the profile measurement creates summaries during every run and appends them to a list. Do we want to have something similar? What I understood from the OP was that we want to print the statistics only at the end of the test?

Reply:
We want to generate a single summary at the end. That said you need to keep track of the information you gather over time - so although slightly different, this will be fairly similar.

Reply:
yeah, i am already keeping track of some of those metrics. just need to surface them out at the end of the tests.

Reply:
hey i have made further changes to the measurements which are inline to the profile measurements and yet generating only a single summary at the end of gather phase.